src/CacheTower/Extensions/ExtensionContainer.cs

```diff
@@ -126,5 +126,23 @@ public async ValueTask DisposeAsync()
 
 			Disposed = true;
 		}
+
+		public void Dispose()
+		{
+			if (Disposed)
+			{
+				return;
+			}
+
+			foreach (var extension in AllExtensions)
+			{
+				if (extension is IDisposable disposable)
```

Comment:
Same issue as the `CacheStack.Dispose` - if an extension only has an asynchronous dispose it won't properly be cleaned up.